gobblin-temporal/src/main/java/org/apache/gobblin/temporal/yarn/YarnService.java

```diff
@@ -590,15 +556,44 @@ protected ByteBuffer getSecurityTokens() throws IOException {
 
   @VisibleForTesting
   protected String buildContainerCommand(Container container, String helixParticipantId, String helixInstanceTag) {
+    long allocationRequestId = container.getAllocationRequestId();
+    // Using getOrDefault for backward-compatibility with containers that don't have allocationRequestId set
+    WorkerProfile workerProfile = this.workerProfileByAllocationRequestId.getOrDefault(allocationRequestId,
+        this.defaultWorkerProfile);
```

Comment:
I see value in recovering, so the app continues onward.  but not having an associated worker profile really is a bug!  let's not silently resolve, but instead:
```
WorkerProfile workerProfile = Optional.ofNullable(this.workerProfileByAllocationRequestId.get(allocationRequestId))
    .orElseGet(() -> {
  log.warn("no WP found for {} ... falling back... ");
  return this.workerProfileByAllocationRequestId.get(DEFAULT_ALLOCATION_REQUEST_ID); // NOTE: instance member unnecessary!
});
```